Patch boringssl for ppc64le

wdirons opened the pull request:
This commit adds a patch file for the bazel BUILD file contained in the
boringssl.tar.gz file. It set the necessary compile options to build boringssl
on pp64le. (Basically adds linux_ppc64le everywhere linux_x86_64 is)

Fixes #20677

wdirons committed:
Patch boringssl for ppc64le

This commit adds a patch file for the bazel BUILD file contained in the
boringssl.tar.gz file. It set the necessary compile options to build boringssl
on pp64le. (Basically adds linux_ppc64le everywhere linux_x86_64 is)

Fixes #20677

wdirons commented:
Related question , if I wanted to upstream this fix so the patch wouldn't be required for Tensorflow where would I do it? The tar.gz file that is downloaded for boringssl includes more files then what I get when cloning https://boringssl.googlesource.com/boringssl/

wdirons commented:
I found that the boringssl repo has a master-with-bazel branch that I need to work from.

gunan commented on third_party/boringssl/boringssl_build_file_ppc64le.patch:
```diff
@@ -0,0 +1,53 @@
+     "crypto_sources",
+     "crypto_sources_linux_x86_64",
+     "crypto_sources_mac_x86_64",
++    "crypto_sources_linux_ppc64le",
```
Previously when we  asked boringssl about such patches, they told us some of these may not work at all.
Can we work with them to make sure this works, and then upstream it into boringssl?

wdirons committed:
Update boringssl dependency for ppc64le/gcc 4.8.5 build failure

In booringssl PR:
https://boringssl-review.googlesource.com/c/boringssl/+/29784

The bazel build file was updated to build ppc64le with the same
compile options as linux_x86_64. This commit updates boringssl
to reference that commit.

jayfurmanek commented:
@gunan Can we get a quick re-review? Looks like @wdirons had to make one last change to resolve https://github.com/tensorflow/tensorflow/issues/20677

Also is it possible to pick this over to the 1.10 branch?

Thx!
